meringue-zest-extension/src/main/java/edu/berkeley/cs/jqf/ZestReplayer.java

```diff
@@ -77,18 +79,28 @@ public boolean hasInput() {
         public void observeGeneratedArgs(Object[] args) {
             if (argumentsDirectory != null) {
                 try {
-                    for (int i = 0; i < args.length; i++) {
-                        File file = new File(argumentsDirectory, String.format("%s.%d", inputFile.getName(), i));
-                        try (PrintWriter out = new PrintWriter(file)) {
-                            out.print(args[i]);
-                        }
-                    }
+                    observeGeneratedArgs(args, inputFile.getName(), 0);
                 } catch (IOException e) {
                     e.printStackTrace();
                 }
             }
         }
 
+        private void observeGeneratedArgs(Object args, String fileName, int depth) throws IOException {
+            if (args.getClass().isArray() && depth < maxRecurDepth) {
+                File file = new File(argumentsDirectory, fileName);
+                file.mkdirs();
+                for (int i = 0; i < Array.getLength(args); i++) {
+                    observeGeneratedArgs(Array.get(args, i), fileName + String.format("/%d", i), depth + 1);
+                }
+            } else {
+                File file = new File(argumentsDirectory, fileName);
+                try (PrintWriter out = new PrintWriter(file)) {
```

Comment:
Should `file` also be in the `try` with resource block so it gets properly closed and doesn't leak file handles?

Reply:
I actually don't know. This is how the original Meringue handles the writing.